codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/ServiceClientConfigurationUtils.java

```diff
@@ -268,6 +301,31 @@ private CodeBlock authSchemeProviderConfigGetter() {
                         .build();
     }
 
+    private CodeBlock customClientConfigParamSetter(String parameterName, String keyName) {
+        return CodeBlock.builder()
+                        .addStatement("$T clientContextParams = config.option($T.CLIENT_CONTEXT_PARAMS)",
+                                      AttributeMap.class, SdkClientOption.class)
+                        .addStatement("config.option($T.CLIENT_CONTEXT_PARAMS, "
+                                      + "clientContextParams.builder().put($T.$N, $N).build())",
```

Comment:
should this be `toBuilder()`, not `builder()`?

Reply:
You validate in the getter that `CLIENT_CONTEXT_PARAMS` is not null. Can it be null? If so, this would cause a NPE.

Reply:
Yeah thats a good catch. Just looking at it again, CLIENT_CONTEXT_PARAMS is never null. I can remove that check in the getter method

Reply:
Yes, I changed it to toBuilder() as well.